code/controllers/stonedmc/subsystem/job.dm

```diff
@@ -97,6 +97,95 @@ SUBSYSTEM_DEF(job)
 	JobDebug("AR has failed, Player: [player], Mind: [player?.mind], Rank: [rank]")
 	return FALSE
 
+/proc/handle_squad(mob/M, rank, latejoin = FALSE)
+	var/strict = FALSE
+	var/datum/squad/P = SSjob.squads[M.client.prefs.preferred_squad]
+	var/datum/squad/R = SSjob.squads[pick(SSjob.squads)]
+	if(M.client?.prefs?.be_special && (M.client.prefs.be_special & BE_SQUAD_STRICT))
+		strict = TRUE
+	switch(rank)
+		if("Squad Marine")
+			if(P && P.assign(M, rank))
```

Comment:
```suggestion
			if(P?.assign(M, rank))
```

Reply:
That's my oldcode, I don't trust your ?. in procs

Reply:
What do you mean you don't trust it? Have you tested it and see it fail?